src/main/java/de/teamlapen/vampirism/tileentity/TotemTileEntity.java

```diff
@@ -75,84 +74,41 @@
 
 @ParametersAreNonnullByDefault
 public class TotemTileEntity extends TileEntity implements ITickableTileEntity, ITotem {
     private static final Logger LOGGER = LogManager.getLogger();
     private static final Random RNG = new Random();
     private static final ResourceLocation nonFactionTotem = new ResourceLocation("none");
-    /**
-     * stores all BoundingBoxes of vampire controlled villages per dimension, mapped from totem positions
-     */
-    private static final HashMap<Dimension, Map<BlockPos, MutableBoundingBox>> vampireVillages = Maps.newHashMap();
-    /**
-     * saves the position
-     */
-    private static final Map<StructureStart, BlockPos> totemPositions = Maps.newHashMap();
-
-    public static boolean isInsideVampireAreaCached(Dimension dimension, BlockPos blockPos) {
-        if (vampireVillages.containsKey(dimension)) {
-            for (Map.Entry<BlockPos, MutableBoundingBox> entry : vampireVillages.get(dimension).entrySet()) {
-                if (entry.getValue().isVecInside(blockPos)) {
-                    return true;
-                }
-            }
-        }
-        return false;
-    }
-
-    public static void clearCacheForDimension(Dimension dimension) {
-        vampireVillages.remove(dimension);
-    }
 
-    private static void addVampireVillage(Dimension dimension, BlockPos pos, MutableBoundingBox box) {
-        vampireVillages.computeIfAbsent(dimension, dimension1 -> Maps.newHashMap()).put(pos, box);
+    @Deprecated
+    public static boolean isInsideVampireAreaCached(Dimension dimension, BlockPos blockPos) { //TODO 1.16 remove
+        return TotemHelper.isInsideVampireAreaCached(dimension, blockPos);
     }
 
-    private static void removeVampireVillage(Dimension dimension, BlockPos pos) {
-        vampireVillages.computeIfPresent(dimension, (dimension1, structureStarts) -> {
-            structureStarts.remove(pos);
-            return structureStarts;
-        });
+    @Deprecated
+    public static void clearCacheForDimension(Dimension dimension) { //TODO 1.16 remove
+        TotemHelper.clearCacheForDimension(dimension);
     }
 
     /**
-     * @return false if another totem exists
-     */
-    private static boolean addTotem(StructureStart structure, BlockPos totemPos) {
-        return totemPositions.computeIfAbsent(structure, (start -> totemPos)).equals(totemPos);
-    }
-
-    private static void removeTotem(StructureStart structure) {
-        totemPositions.remove(structure);
-    }
-    /**
-     * @return {@code null} if no totem exists
+     * this method has no use anymore, because the village now persists of a collection of {@link PointOfInterest} and not based on a structure start
+     *
+     * @throws Exception
      */
+    @Deprecated
     public static @Nullable
-    BlockPos getTotemPosition(StructureStart structure) {
-        return totemPositions.get(structure);
+    BlockPos getTotemPosition(StructureStart structure) throws Exception { //TODO 1.16 remove
```

Comment:
@maxanier this method will not work with the new system. Should i be kept like this, or can it be removed or return null?

Reply:
Is there a replacement for the current use cases?
Since it is not exposed via an old API, I don't see a reason why we should keep it

Reply:
yes in TotemHandler with param `Collection<PointOfInterest>`.
Then i remove it